### PR TITLE
[BEAM-1542] Takeover #2166 with initial support for SpannerIO.Write

### PR DESCRIPTION
**Pull request by dhalperi**
This takes over #2166, and it is not really ready for use. To be properly ready, it needs:

* Many style guide fixes.
* Comprehensive javadoc.
* Unit tests and integration tests.

However, given that 2.0.0 was just cut and there is a month until next release, I believe it should be okay to merge this as @mairbek and @NextDevBoard are actively working on it.

We will pull it from the 2.1.0 release if these major issues are not addressed before then.

R: @davorbonaci 
CC: @mairbek @nextdevboard

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java [hunk 1]
```diff
@@ -75,6 +75,8 @@ public void testGcpApiSurface() throws Exception {
             classesInPackage("javax"),
             classesInPackage("org.apache.beam"),
             classesInPackage("org.apache.commons.logging"),
+            classesInPackage("com.google.cloud"),
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java]: This needs work.

**Comment by mairbek** [comment]
Thank you for updates, one thing is that we need to bump grpc-google-common-protos.version to 0.1.9, otherwise Spanner client fails with a runtime error.

**Comment by dhalperi** [comment]
The proper way to protect against such issues is to have tests, both unit
and integration!

**Comment by dhalperi** [comment]
R: @mairbek 
CC: @nextdevboard fyi -- getting this going.

**Comment by mairbek** [comment]
LGTM

**Pull request closed without merging**